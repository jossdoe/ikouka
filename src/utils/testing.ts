// This is a proxy-file to allow for global wrappers and modifications
// to the functionality of React Testing Library.
// https://testing-library.com/docs/react-testing-library/setup#custom-render
export * from '@testing-library/react'
