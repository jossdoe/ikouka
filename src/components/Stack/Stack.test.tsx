import React from 'react'
import { render, screen } from '../../utils/testing'
import { Stack, StackProps } from './index'

const SIZES: StackProps['size'][] = ['xs', 'sm', 'md', 'lg', 'xl', undefined]

SIZES.forEach((size) => {
  test(`Stack - ${size}`, () => {
    const { asFragment } = render(
      <Stack size={size}>
        <button />
        <button />
        <button />
      </Stack>
    )

    expect(asFragment()).toMatchSnapshot()

    expect(screen.getAllByRole('button')).toHaveLength(3)
  })
})
