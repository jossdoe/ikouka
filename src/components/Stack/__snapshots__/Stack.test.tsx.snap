// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack - lg 1`] = `
<DocumentFragment>
  <div
    class="space-y-6 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Stack - md 1`] = `
<DocumentFragment>
  <div
    class="space-y-4 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Stack - sm 1`] = `
<DocumentFragment>
  <div
    class="space-y-2 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Stack - undefined 1`] = `
<DocumentFragment>
  <div
    class="space-y-4 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Stack - xl 1`] = `
<DocumentFragment>
  <div
    class="space-y-8 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;

exports[`Stack - xs 1`] = `
<DocumentFragment>
  <div
    class="space-y-1 "
  >
    <button />
    <button />
    <button />
  </div>
</DocumentFragment>
`;
